Scripts/UI/Lobby/Network/server_info_and_chat.gd

```diff
@@ -135,3 +158,20 @@ func _on_button_poll_ip_pressed():
 	var external_ip = await NET.fetch_external_address_guess()
 	if NET.server:
 		NET.server.server_external_address = external_ip
+		external_ip_edit.text = external_ip
+
+
+func _on_is_public_check_box_toggled(toggled_on):
+	if toggled_on:
+		if NET.server.server_external_address == "-needs fetch-":
+			await _on_button_poll_ip_pressed()
```

Comment:
this should be called only after configuration -- if no configuration is done, this checkbox should redirect us to settings with it